### DISCUSSION TITLE
Making predictions on panel data with differing date times

### DISCUSSION THREAD

**Discussion by DurojaiyeAbisoye** [alan-turing-institute/sktime#6356] [General]
Hello, I'm new to this framework, and I'm trying to us it for panel data forecasting. I've arranged my data as described here: https://sktime-backup.readthedocs.io/en/stable/examples/01_forecasting.html#1.2.6-Panel-forecasts-and-hierarchical-forecasts, and end up with data that has Company ID, and Date as the multi-index columns. I then used the temporal train test split to split data, and I end up with data and train a random forest regressor using make_reduction as described here: https://sktime-backup.readthedocs.io/en/stable/examples/01c_forecasting_hierarchical_global.html#Rationale-for-tree-based-models. The issue is I get this error: NotImplementedError: Generating in-sample predictions is not yet implemented for RecursiveTabularRegressionForecaster. Personally, I think this is because the test data I created, has different Company IDs, and dates that I'm trying to forecast for. Is there any workaround this?

**Comment by fkiraly** [comment]
@DurojaiyeAbisoye, the default reducer may not be able to cope with unequal index sets.
You could try `DirectReductionForecaster` instead.

What would also be helpful is the code and your versions.

You are also linking to the `sktime` backup server which hasn´t been updated in a while - we only use these in case the primary docs server is down (which hasn´t happened in a year). May I ask, where do you have these links from?
